src/components/Sidebar.js

```diff
@@ -15,16 +24,38 @@ class Sidebar extends Component {
     }
   }
 
+  componentWillReceiveProps (nextProps) {
+    this.setState({
+      items: nextProps.stacks.items
+    })
+  }
+
+  filterStacks (e) {
+    const query = e.currentTarget.value
+    const $items = this.props.stacks.items.filter((item) => {
+      if (item.name.indexOf(query) > -1) {
+        return item.name
+      }
```

Comment:
`.filter` just should return true or false. So you can simplify this using `return item.name.indexOf(query) > -1`